### PR TITLE
support object store

### PR DESCRIPTION
**Pull request by yihuang**

### CODE REVIEW

File: memdb.go [hunk 1]
```diff
@@ -1,173 +1,195 @@
...
-}
+type (
+	MemDB    = GMemDB[[]byte]
+	ObjMemDB = GMemDB[any]
+)
 
-var _ KeyItem = memdbItem{}
+var (
+	_ storetypes.KVStore = (*MemDB)(nil)
+	_ storetypes.KVStore = (*MemDB)(nil)
```
Comment (yihuang) [inline memdb.go:16]:
```suggestion
	_ storetypes.ObjKVStore = (*ObjMemDB)(nil)
```

File: mvdata.go [hunk 2]
```diff
@@ -125,44 +180,55 @@ func (d *MVData) SnapshotTo(cb func(pair KVPair) bool) {
...
-	Value       []byte
+	Value       V
 	Estimate    bool
 }
 
-func secondaryLesser(a, b secondaryDataItem) bool {
-	// reverse the order
-	return a.Index > b.Index
+func secondaryLesser[V any](a, b secondaryDataItem[V]) bool {
+	return a.Index < b.Index
```
[the remark this answers is not in the document]
Comment (yihuang) [reply mvdata.go:224]: First of all, ascending order is more intuitive, secondly the iterator api use more memory allocations, the speed difference is marginal though if any at all.

[the remark this answers is not in the document]
Comment (mmsqe) [reply mvdata.go:224]: oh right, no need handle Release

File: types.go [hunk 3]
```diff
@@ -55,16 +52,7 @@ type ReadSet struct {
...
-
-func NewWriteSet() WriteSet {
-	return *NewMemDBNonConcurrent()
-}
-
-type (
-	MultiWriteSet = []WriteSet
-	MultiReadSet  = []ReadSet
-)
+type MultiReadSet = map[int]*ReadSet
```
Comment (mmsqe) [inline types.go:55]: Can reduce size since it was possible to skip index in the middle before right?

**Review by mmsqe — APPROVED** [review]

File: types.go (hunk 3 above)
Comment (yihuang) [reply types.go:55]: exactly, can save a little bit

**Pull request merged**